library/log.go

```diff
@@ -36,6 +42,47 @@ func (l *Log) AppendData(data []byte) {
 	l.SetData(append(l.GetData(), data...))
 }
 
+// MaskData reads through the log data and masks
+// all values provided in the string slice. If the
+// log is empty, we do nothing.
+func (l *Log) MaskData(secrets []string) {
```

Comment:
Thanks for your work on this 🎉 

My first piece of feedback would be to remove most, if not all, `[]byte` -> `string` conversions.

Unless they are required, this should increase the performance by removing those conversions.

i.e. 

* `strData := string(l.GetData())` -> `data := l.GetData()`
* `matches := re.FindAllString(strData, -1)` -> `matches := re.FindAll(data, -1)`
* `strData = strings.Replace(strData, match, mask, 1)` -> `data = bytes.Replace(data, match, mask, 1)`

However, I'm not an expert with regexp but I wonder if we need to construct a new mask for each match?

Feels like this could be accomplished with the regexp library without extra logic.

Here is a sample of what I mean:

https://go.dev/play/p/vm4IAGT-IUp

A rough example of how this might look in our code:

```go
	data := l.GetData()

	for _, secret := range secrets {
		// escape all regexp metacharacters that exist in the secret
		//
		// https://pkg.go.dev/regexp#QuoteMeta
		escaped := regexp.QuoteMeta(secret)

		// create regexp to match secrets in the log data surrounded by regexp metacharacters
		//
		// https://pkg.go.dev/regexp#MustCompile
		re := regexp.MustCompile(`(\s|^|"|:|'|\.|,)` + escaped + `(\s|$|"|:|'|\.|,)`)

		// create a mask for the secret
		mask := fmt.Sprintf("$1%s$2", constants.SecretLogMask)

		// replace all regexp matches of secret with mask
		//
		// https://pkg.go.dev/regexp#Regexp.ReplaceAll
		data = re.ReplaceAll(data, []byte(mask))

		// update log data with any secret values masked
		l.SetData(data)
	}
```

Reply:
Yes -- this seems to work and is much cleaner. Thanks!